File: hyezuu/PGS/5명씩.java
class Solution {
    public String[] solution(String[] names) {
        int len = names.length;
        String[] answer = new String[len/5 + (len%5==0 ? 0 : 1)];
        int idx = 0;
        for(int i=0; i<len; i+=5){
            answer[idx++] = names[i];
        }
        return answer;
    }
}

File: ukidelly/PGS/240608/5명씩.kt
class Solution {
  fun solution(names: Array<String>): Array<String> {
    return names.slice(0..names.lastIndex step 5).toTypedArray()
  }

  fun solution2(names: Array<String>): Array<String> {
    return names.toList().chunked(5) { it.first() }.toTypedArray()
  }
}